typescript/23-hello-interview-trees/examples.ts

```diff
@@ -0,0 +1,103 @@
+import { TreeNode } from "./trees.ts";
+
+/*
+  Given the root of a binary tree and an integer target,
+  write a recursive function to determine if the tree has
+  a root-to-leaf path where all the values along that path sum to the target.
+*/
+
+export function hasPathSum(root: TreeNode | null, target: number): boolean {
+  if (!root) {
+    return false;
+  }
+
+  if (!root.left && !root.right) {
+    return target === root.val;
+  }
+
+  const left = hasPathSum(root.left, target - root.val);
+  const right = hasPathSum(root.right, target - root.val);
+  return left || right;
+}
+
+/*
+  Given the root of a binary, write a recursive function
+  to determine if it is a valid binary search tree.
+*/
+
+export function validateBST(root: TreeNode | null): boolean {
+  const dfs = (node: TreeNode | null, min: number, max: number): boolean => {
```

Comment:
It's a recursive traversal routine, not DFS. DFS stands for depth-first **search**, and we aren't searching for anything. While this shares the skeleton with DFS, naming is confusing.